docs/guide/api-javascript.md

```diff
@@ -189,12 +197,90 @@ import { preview } from 'vite'
 async function resolveConfig(
   inlineConfig: InlineConfig,
   command: 'build' | 'serve',
-  defaultMode?: string
+  defaultMode = 'development'
 ): Promise<ResolvedConfig>
 ```
 
 The `command` value is `serve` in dev (in the cli `vite`, `vite dev`, and `vite serve` are aliases).
 
+## `mergeConfig`
+
+**Type Signature:**
+
+```ts
+function mergeConfig(
+  defaults: Record<string, any>,
+  overrides: Record<string, any>,
+  isRoot = true
+): Record<string, any>
+```
+
+Deeply merge two Vite configs. `isRoot` represents the level within the Vite config which is being merged. For example, set `false` if you're merging two `build` options.
+
+## `searchForWorkspaceRoot`
+
+**Type Signature:**
+
+```ts
+function searchForWorkspaceRoot(
+  current: string,
+  root = searchForPackageRoot(current)
+): string
+```
+
+**Related:** [server.fs.allow](/config/server-options.md#server-fs-allow)
+
+Search for the root of the potential workspace if it meets the following conditions, otherwise it would fallback to `root`:
+
+- contains `workspaces` field in `package.json`
+- contains one of the following file
+  - `lerna.json`
+  - `pnpm-workspace.yaml`
+
+## `loadEnv`
+
+**Type Signature:**
+
+```ts
+function loadEnv(
+  mode: string,
+  envDir: string,
+  prefixes: string | string[] = 'VITE_'
+): Record<string, string>
+```
+
+**Related:** [`.env` Files](./env-and-mode.md#env-files)
+
+Load `.env` files within the `envDir`. By default only env variables prefixed with `VITE_` are loaded, unless `prefixes` is changed.
+
+## `normalizePath`
+
+**Type Signature:**
+
+```ts
+function normalizePath(id: string): string
+```
+
+**Related:** [Path Normalization](./api-plugin.md#path-normalization)
+
+Normalizes a path to interoperate between Vite plugins.
+
+## `optimizeDeps`
+
+**Type Signature:**
+
+```ts
+async function optimizeDeps(
+  config: ResolvedConfig,
+  force = config.optimizeDeps.force,
+  asCommand = false
+): Promise<DepOptimizationMetadata>
+```
+
+**Related:** [Dependency Pre-Bundling](/guide/dep-pre-bundling)
+
+Scan and optimize dependencies within a project. Used by Vite CLI when running `vite optimize`.
```

Comment:
We may want to review also this API, as it now only covers optimizing some cases. For example, SSR deps optimization isn't supported. I wonder if someone is using this function in the wild? Is it for pre-warm Vite?

Reply:
I kept this mainly to be consistent with Vite's CLI commands, but yeah it seems to be getting niche to use this directly. I used this in vite-plugin-svelte at one point for `optimizeDeps.extensions` but removed now as I used `optimizeDeps.force` instead. Let's remove this too.